src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -86,19 +87,28 @@ public InnReachResponseDTO trackShippedItem(String trackingId, String centralCod
     var innReachTransaction = getTransaction(trackingId, centralCode);
 
     var itemBarcode = itemShipped.getItemBarcode();
+    var callNumber = itemShipped.getCallNumber();
 
     var transactionPatronHold = (TransactionPatronHold) innReachTransaction.getHold();
-    transactionPatronHold.setShippedItemBarcode(itemBarcode);
 
-    if (Objects.nonNull(itemBarcode)) {
-      var itemByBarcode = inventoryService.getItemByBarcode(itemBarcode);
-      if (Objects.nonNull(itemByBarcode)) {
+    if (nonNull(itemBarcode)) {
+      transactionPatronHold.setShippedItemBarcode(itemBarcode);
+
+      inventoryService.findItemByBarcode(itemBarcode).ifPresentOrElse(item -> {
         transactionPatronHold.setShippedItemBarcode(itemBarcode + transactionPatronHold.getItemAgencyCode());
-      }
+        transactionPatronHold.setFolioItemBarcode(itemBarcode + transactionPatronHold.getItemAgencyCode());
```

Comment:
If I'm not mistaken `itemAgencyCode` should be appended only to `folioItemBarcode`.

Reply:
Update transaction itemBarcode shippedItemBarcode with request payload itemBarcode value (if present)

- If itemBarcode already exists in FOLIO inventory, append the item's agencyCode

from https://issues.folio.org/browse/MODINREACH-87

Reply:
@andrew-khodyka how about this?) https://issues.folio.org/browse/MODINREACH-87?focusedCommentId=116706&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-116706

> The itemBarcode from the request payload should be stored on the transaction record as the shippedItemBarcode and then we should attempt to use this barcode as the barcode for the FOLIO item. If the barcode already exists in FOLIO, append the item's agency code to the barcode and **save that value to the FOLIO item record as the barcode and set the transaction's folioItemBarcode value to this updated value**. If the agency code is not appended, the itemBarcode value supplied in the request payload will also be used for folioItemBarcode on the transaction record.

Reply:
Also, there is a discussion in related task https://issues.folio.org/browse/MODINREACH-150?focusedCommentId=116478&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-116478